### DISCUSSION TITLE
Exclude PV of backup (Multiple CSI-based storage classes)

### DISCUSSION THREAD

**Discussion by headyj** [vmware-tanzu/velero#3628] [Community support Q&A]
Hello,

I'm using the great [CSI plugin](vmware-tanzu/velero-plugin-for-csi) for backing-up my [EBS CSI](https://github.com/kubernetes-sigs/aws-ebs-csi-driver) disks. Problem is I'm also using  [EFS CSI driver](https://github.com/kubernetes-sigs/aws-efs-csi-driver) for other disks. According to [this issue](https://github.com/vmware-tanzu/velero/issues/2999#issuecomment-715453058), it's currently not possible to use both restic (For EFS) and standard plugin (For EBS) at the same time.

To mitigate this limitation, my idea is to use [AWS backup](https://docs.aws.amazon.com/efs/latest/ug/awsbackup.html) for EFS disks, and stick with Velero for EBS disks. The problem is that I cannot find a way to "exclude" EFS disks from the backup:
* If I don't create the VolumeSnapshotClass for EFS, My backup partially fails (which is normal):
`failed to get volumesnapshotclass for storageclass efs-storage-class: failed to get volumesnapshotclass for provisioner efs.csi.aws.com, ensure that the desired volumesnapshot class has the velero.io/csi-volumesnapshot-class label`
* If I create a volumesnapshotclass for my EFS volume, it also fails as I thing he tries to create a snapshot for EFS volume, which is not possible.

Is there any way to tell Velero to just exclude PVs based on some annotations or labelling?

Thank you in advance :)

**Comment by laimison** [comment]
I have related question too. Is there a way to exclude whole `storageclass` so all PV/PVCs under it are excluded?

```
$ k get sc --no-headers | grep local-hostpath
local-hostpath              openebs.io/local                                           Retain   WaitForFirstConsumer   false   92d
```

So far the work around is to label each new PVCs once new is created, but it requires work to take care of backup once new PVC is created. Did I miss right solution?

I use this command to ensure PVCs are labeled so excluded from a backup:

```
k -n percona-l label pvc datadir-my-percona-cluster-pxc-0 velero.io/exclude-from-backup=true
```
